src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs

```diff
@@ -255,7 +276,9 @@ protected override void Write(AsyncLogEventInfo logEvent)
         {
             this.MergeEventProperties(logEvent.LogEvent);
             this.PrecalculateVolatileLayouts(logEvent.LogEvent);
-            this.RequestQueue.Enqueue(logEvent);
+            bool queueWasEmpty = this.RequestQueue.Enqueue(logEvent);
+            if (queueWasEmpty && TimeToSleepBetweenBatches <= 0)
```

Comment:
Yes as the RequestQueue holds a lock, when returning whether queue was empty before the enqueue. A concurrent thread will get a different result.